app/src/main/java/com/tiptopgoodstudio/androidresources/viewmodel/ResourceViewModel.java

```diff
@@ -3,95 +3,40 @@
 import android.app.Application;
 import android.arch.lifecycle.AndroidViewModel;
 import android.arch.lifecycle.LiveData;
-import android.arch.lifecycle.MutableLiveData;
-import android.os.AsyncTask;
 import android.support.annotation.NonNull;
 
 import com.tiptopgoodstudio.androidresources.ResourceRepository;
 import com.tiptopgoodstudio.androidresources.db.entity.Resource;
 
 import java.util.List;
 
-/**
- * The following added by Divya on 4/20/2018
- * Added an AsyncTask method to call the Repository method. This is to ensure
- * that the database call is not being made on the main thread
- *
- */
-
 public class ResourceViewModel extends AndroidViewModel {
 
     private ResourceRepository mResourceRepository;
-    private MutableLiveData<List<Resource>> mResources = new MutableLiveData<List<Resource>>();
-    private MutableLiveData<List<Resource>> mTopicResources = new MutableLiveData<List<Resource>>();
+    private LiveData<List<Resource>> mAllResources;
+    private LiveData<List<Resource>> mTopicResources;
 
     public ResourceViewModel(@NonNull Application application) {
         super(application);
-        mResourceRepository = new ResourceRepository(application);
-        //upon creation, ViewModel caches list of all resources
-        loadAllResources();
+        mResourceRepository = ResourceRepository.getRepository(application);
+        mAllResources = mResourceRepository.getAllResourcesList();
     }
 
     /**
      * Return the cached member variable for all resources
      */
     public LiveData<List<Resource>> getAllResources() {
-        return mResources;
+        return mAllResources;
     }
 
     /**
-     * When this method is observed from the View, the AsyncTask queries the
-     * resource_table based on the given topic. LiveData ensures that after the
-     * background task is completed, the data is propagated to the UI
-     *
+     * @param topic
+     * @return LiveData<List < Resource>> filtered by topic
      */
     public LiveData<List<Resource>> getTopicResources(String topic) {
-        new GetTopicResourcesAsyncTask(mResourceRepository).execute(topic);
```

Comment:
The reason for having AsyncTasks to call the ResourceRepository methods was because Room does not allow accessing the database on the main thread unless you use allowMainThreadQueries() in the Room AppDatabase creation. Since you did not seem to add that, I wonder how this code ran without the runtime error generated by Room.

Reply:
Correct.  The db cannot be queried on the main thread.  With Room, however, since the DAO method specified that a LiveData<> object is returned, that query is automatically performed on a background thread behind the scenes (you don't have to manually create a thread - Android does it for you).  If we were doing any other db transaction or if the results returned from the db were not an instance of LiveData but were instead wrapped in a LiveData wrapper later (like you did in the ViewModel) then it WOULD need to be done using a separate thread.  This is one of the beautiful things / quirks about returning a LiveData instance from a Room db query when it is defined in the DAO as such.  I'm trying to find references, but I have so many saved in Evernote on the topic that I will need to send that to you on Slack once I find it.

Reply:
This is a great find! Definitely reduces a lot of boilerplate code :) Awesome!

Reply:
It simplified the code to such an extent I was jumping for joy!  :-D  
The interesting thing is that you can't return a MutableLiveData<> instance from the DAO.  I have searched for why that is but have not found an answer yet.  This was the problem the first time I implemented Room - trying to return MutableliveData.   

Since the ResourceListActivity creates an instance of ResourceViewModel each time it is created, (and both are destroyed when the activity is destroyed), this made retrieving a fresh LiveData instance returning subject-specific query results a super slick solution.  If we decide to switch to a single activity with multiple fragments, this may involve a transformation, but one issue at a time, yes?